### ISSUE TITLE
Spurious choice-points in frame.pl ?

### ISSUE THREAD

**Issue by pmoura** [terminusdb/terminusdb#50]
Are the implicit choice-points created by the following `member/2` and `select/3` calls required?

https://github.com/terminusdb/terminus-server/blob/df1ba67477fd0d32f9f4062eb18e62e2982b2117/library/frame.pl#L818-L819

The call to the `realise_triples/4` predicate from the `object_edges/3` predicate suggests that those calls could be replaced with `memberchk/2` and `selectchk/3`. As the `realise_triples/4` predicate is recursive, these choice-points accumulate.

P.S. Given `sort/2` semantics, the `object_edges/3` predicate can also be rewritten as:

```prolog
object_edges(URI,Database,Edges) :-
    (   most_specific_type(URI,Class,Database),
        class_frame(Class,Database,Frame),
        realise_triples(URI,Frame,Database,Unsorted)
    ->  sort(Unsorted,Edges)
    % There is no type in the database, so it doesn't exist...
    ;   Edges=[]).
```

**Comment by GavinMendelGleason** [comment]
The choice points are indeed spurious. This entire section should be rewriten and the object_edges version looks better as well.

**Comment by GavinMendelGleason** [comment]
Fixed in e28b1f4
